docs/Inside_Arbitrum_Nitro/Inside_Arbitrum_Nitro.md

```diff
@@ -0,0 +1,527 @@
+
+Because the protocol doesn't trust the sequencer not to put garbage into its sequence, the state transition function will detect and discard any invalid transactions in the sequence. A well-behaved sequencer will filter out invalid transactions so the state transition function never sees them--and this reduces cost and therefore keeps transactions fees low--but Nitro will still work correctly no matter what the sequencer puts into its feed. (Transactions in the feed are signed by their senders, so the sequencer can't create forged transactions.)
+
+The state transition function is deterministic, which means that its behavior depends only on the current state and the contents of the next transaction--and nothing else. Because of this determinism, the result of a transaction T will depend only on the genesis state of the chain, the transactions before T in the sequence, and T itself. 
+
+It follows that anyone who knows the transaction sequence can compute the state transition function for themselves--and all honest parties who do this are guaranteed to get identical results. This is the normal way that Nitro nodes operate: get the transaction sequence, and run the state transition function locally. No consensus mechanism is needed for this.
+
+### How the Sequencer Publishes the Sequence
+
+So how do nodes get the sequence? The sequencer publishes it in two ways: a real-time feed, and batches posted on L1 Ethereum.
+
+The real-time feed is published by the sequencer so that anyone who subscribes to the feed receives instant notification of each transaction as it is sequenced. Nitro nodes can subscribe to the feed directly from the sequencer, or through a relay that forwards the feed. The feed represents the sequencer's promise that it will record transactions in a particular order. If the sequencer is honest and doesn't have a long downtime, this promise will be kept. So anyone who trusts the sequencer to keep it promises can rely on the feed to get instant information about the transaction sequence--and they can run the sequenced transactions through the state transition function to learn the results of each transaction immediately. This is "soft finality" for transactions; it's "soft" because it depends on the sequencer keeping its promises.
+
+The sequencer also publishes its sequence on the L1 Ethereum chain. Periodically--perhaps every few minutes in production--the sequencer concatenates the next group of transactions in the feed, compresses them for efficiency, and posts the result as calldata on Ethereum. This is the final and official record of the transaction sequence. As soon as this Ethereum transaction has finality on Ethereum. the Layer 2 Nitro transactions it records will have finality. These transactions are final because their position in the sequence has finality, and the outcome of the transactions is deterministic and knowable to any party. This is "hard finality".
```

Comment:
```suggestion
The sequencer also publishes its sequence on the L1 Ethereum chain. Periodically--perhaps every few minutes in production--the sequencer concatenates the next group of transactions in the feed, compresses them for efficiency, and posts the result as calldata on Ethereum. This is the final and official record of the transaction sequence. As soon as this Ethereum transaction has finality on Ethereum, the Layer 2 Nitro transactions it records will have finality. These transactions are final because their position in the sequence has finality, and the outcome of the transactions is deterministic and knowable to any party. This is "hard finality".
```